control/veda_file_discovery.py

```diff
@@ -125,162 +127,199 @@ def about_video_validate(self, meta, key):
-        )
-        ingest.insert()
-
-        if ingest.complete is False:
-            return
+        return transcript_preferences
 
+    def discover_studio_ingested_videos(self):
         """
-        Delete Original After Copy
+        Discovers studio ingested videos, for further validations and processes.
         """
-        if self.auth_dict['edx_s3_ingest_prefix'] is not None:
-            new_key = 'prod-edx/processed/' + key.name[::-1].split('/')[0][::-1]
-            key.copy(self.bucket, new_key)
+        if self.node_work_directory:
+            try:
+                connection = boto.connect_s3()
+                self.bucket = connection.get_bucket(self.auth_dict['edx_s3_ingest_bucket'])
+                for video_s3_key in self.bucket.list(self.auth_dict['edx_s3_ingest_prefix'], '/'):
+                    self.validate_and_feed_to_ingest(video_s3_key=self.bucket.get_key(video_s3_key.name))
+            except S3ResponseError:
+                ErrorObject.print_error(message='[File Ingest] S3 Ingest Connection Failure')
+        else:
+            ErrorObject.print_error(message='[File Ingest] No Working Node directory')
 
-        key.delete()
+    def validate_and_feed_to_ingest(self, video_s3_key):
+        """
+        Validates the video key and feed it to ingestion phase.
 
+        Arguments:
+            video_s3_key: An S3 Key associated with a (to be ingested)video file.
 
-def main():
-    pass
+        Process/Steps:
+            1 - Get or create an associated course for a video.
+            2 - Download video to node working directory from S3.
+            3 - Check if this video has valid 3rd Party transcript provider along with the preferences.
+            4 - Set up an ingest instance and insert video to ingestion phase.
+            5 - On completing ingestion, mark the video file as processed.
 
+            Note:
+                Failure at any discovery point will cause video file to be marked as rejected.
+        """
+        client_title = video_s3_key.get_metadata('client_video_id')
+        course_hex = video_s3_key.get_metadata('course_video_upload_token')
+        course_id = video_s3_key.get_metadata('course_key')
+        transcript_preferences = video_s3_key.get_metadata('transcript_preferences')
+        filename = os.path.basename(video_s3_key.name)
+
+        # Try getting course based on the S3 metadata set on the video file.
+        course = self.get_course(course_id, course_hex=course_hex)
+        if course:
+            # Download video file from S3 into node working directory.
+            file_extension = os.path.splitext(client_title)[1][1:]
+            file_downloaded = self.download_video_to_working_directory(video_s3_key, filename, file_extension)
+            if not file_downloaded:
+                # S3 Bucket ingest failed, move the file rejected directory.
+                self.move(video_s3_key, destination_dir=self.auth_dict['edx_s3_rejected_prefix'])
```

Comment:
Here we are just moving the file to rejected directory. Don't we need to update the status on edxval?

Reply:
It was not doing that before so just simulating the previous behavior